server/resources/CampaignResources.py

```diff
@@ -52,26 +53,49 @@ class CreateStepToCampaign(Resource):
     def post(self, id):
         data = campaignStepsParser.parse_args()
         campaign = CampaignModel.find_by_id(id)
+        template_id = data['id']
         if not campaign:
             return {'message': 'Campaign {} doesn\'t exist'.format(id)}, 400
-        new_email_template = EmailTemplateModel(
-            type = data["type"],
-            subject = data["subject"],
-            body = data["body"]
-        )
+        if campaign.owner_id != get_jwt_identity():
+            return {'message': 'You don\'t have permission to do that.'}, 403
+        # new_email_template = EmailTemplateModel(
+        #     name = data["name"],
+        #     type = data["type"],
+        #     subject = data["subject"],
+        #     body = data["body"]
+        # )
         new_step = StepModel(
             campaign_id = id,
-            email_template = new_email_template  
+            email_template_id = template_id 
         )
         try:
-            new_email_template.save_to_db()
-            new_step.save_to_db
+            new_step.save_to_db()
             return {
-                'message': 'Successfully created Step {} for Campaign {}'.format(new_step.id, campaign.name)
+                'step' : new_step.to_dict(rules = ('-email_template.steps', '-campaign'))
```

Comment:
to_dict is a method from a sql serializer library that I am using to serialize db Models, you guys can find its documentation here.

https://github.com/n0nSmoker/SQLAlchemy-serializer

Reply:
Notice I set some rules in the method, otherwise it will create a Recursion infinite loop because of the db relationships and backref. For example: campaign->steps->campaign... so on.